### PR TITLE
Fix error message to fit desired usage for indexing of integers larger than max int

### LINKED ISSUE #116: [PE-D] Using an integer above/below MAX_INT for `get` command results in unexpected error message
How to reproduce: enter `get 2147483647832`

Actual:

Expected:

Likewise for -MAX_INT

-------------
Labels: `type.FunctionalityBug` `severity.Medium`
original: CraveToCode/ped#1

### PR DESCRIPTION
**Pull request by ykwei7**
Resolves #116, #123, #130, #135, #142

Resolves get, remark, deletelesson, edit and clearremark

Adjusted ParserUtil to break down indexing into 2 cases:
1. When input is valid integer and is larger than max_int
2. When input is non-valid

**When input < max_int**
image
**When input > max_int**
image
**When input wiithin list range**
image
**When input is empty**
image

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 1]
```diff
@@ -36,16 +37,59 @@
...
+        String removedLeadingZeros = trimmedIndex.substring(currIndex);
+        return !(removedLeadingZeros.length() > 9);
+    }
+
+    /**
+     * Returns true is string contains all digits from 0 to 9
+     * @param s
+     * @return
+     */
+    private static boolean isAllDigits(String s) {
```
Comment (botr99) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:67]: I think a regex check suffices here. `return s.matches("^\\d+$")`

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 2]
```diff
@@ -36,16 +37,59 @@
...
      * Parses {@code oneBasedIndex} into an {@code Index} and returns it. Leading and trailing whitespaces will be
      * trimmed.
      * @throws ParseException if the specified index is invalid (not non-zero unsigned integer).
      */
-    public static Index parseIndex(String oneBasedIndex) throws ParseException {
+    public static Index parseIndex(String oneBasedIndex) throws ParseException, IndexOutOfBoundsException {
         String trimmedIndex = oneBasedIndex.trim();
-        if (!StringUtil.isNonZeroUnsignedInteger(trimmedIndex)) {
+
+        if (!isAllDigits(trimmedIndex) || trimmedIndex.equals("0") || trimmedIndex.equals("")) {
```
Comment (botr99) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:85]: Does `isAllDigits` cover the case whereby `trimmedIndex.equals("")`?

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 3]
```diff
@@ -36,16 +37,59 @@
 
     public static final String MESSAGE_INVALID_INDEX = "Index is not a non-zero unsigned integer.";
 
+    public static final String MESSAGE_INDEX_OUT_OF_BOUNDS = "The index provided is invalid";
```
Comment (botr99) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:40]: Very minor nitpick, remember to add in a full stop here 😅

File: src/main/java/seedu/address/ui/MainWindow.java [hunk 4]
```diff
@@ -191,6 +194,10 @@ private CommandResult executeCommand(String commandText) throws CommandException
             logger.info("Invalid command: " + commandText);
             resultDisplay.setFeedbackToUser(e.getMessage());
             throw e;
+        } catch (IndexOutOfBoundsException ie) {
+            logger.info("Invalid index: " + commandText);
```
Comment (botr99) [inline src/main/java/seedu/address/ui/MainWindow.java:198]: Good use of logging. 👍

**Review by botr99 — APPROVED** [review]

File: src/main/java/seedu/address/logic/parser/ParserUtil.java (hunk 1 above)
Comment (ykwei7) [reply src/main/java/seedu/address/logic/parser/ParserUtil.java:67]: Thanks for the suggestion! Have replaced the existing function with this regex.

File: src/main/java/seedu/address/logic/parser/ParserUtil.java (hunk 2 above)
Comment (ykwei7) [reply src/main/java/seedu/address/logic/parser/ParserUtil.java:85]: Have removed the unnecessary condition, thanks for pointing out!

File: src/main/java/seedu/address/logic/parser/ParserUtil.java (hunk 3 above)
Comment (ykwei7) [reply src/main/java/seedu/address/logic/parser/ParserUtil.java:40]: Okay thanks for the keen observation!

**Pull request merged**